openapi.go

```diff
@@ -97,9 +97,28 @@ func (c *oaComponents) addSchema(t reflect.Type, mode schema.Mode, hint string,
 		// TODO: See if this type has a predefined schema associated with it
 		// via a predefined map, a callback or an interface.
 		var err error
-		if s, err = schema.GenerateWithMode(t, mode, nil, map[string]string{}); err != nil {
+		nestedSchemas := map[string]schema.NestedSchemaReference{}
+		if s, err = schema.GenerateWithMode(t, mode, nil, nestedSchemas); err != nil {
 			panic(err)
 		}
+		// Iterate over any nested schemas that are referenced by the schema
+		// we just generated
+		for k, v := range nestedSchemas {
+			// Check to see if this component already has a definition for that
+			// type in its schema collection
+			_, exists := c.Schemas[k]
+			if !exists {
+				// If not, generate the schema for the nested schema.
+				predefined := map[string]schema.NestedSchemaReference{}
+				s, err := schema.GenerateWithMode(v.Type, mode, nil, predefined)
+				if err != nil {
+					panic(err)
+				}
+				// Add the generated schema to the list of schemas associatd
+				// with this component.
+				c.Schemas[k] = s
```

Comment:
Found the fix. There is logic in `addExistingSchema` that wasn't being run, causing a later `DeepEqual` to fail and generating extra models. With this change the tests pass again for me.

```suggestion
				c.addExistingSchema(s, k, generateSchemaField)
```

Reply:
@mt35-rs let me know if you want to apply this, otherwise I can open a new PR with the fix.

Reply:
Probably better for you to apply this.  You know the ins and outs better than I do.